tests/components/azure_event_hub/test_init.py

```diff
@@ -90,128 +38,173 @@ async def test_full_config(hass):
+            },
+            [
+                FilterTest("climate.excluded", False),
+                FilterTest("light.included", True),
+                FilterTest("sensor.excluded_test", False),
+                FilterTest("sensor.included_test", True),
+                FilterTest("binary_sensor.included", True),
+                FilterTest("binary_sensor.excluded", False),
+            ],
+        ),
+        (
+            {
+                "exclude_domains": ["climate"],
+                "exclude_entity_globs": ["sensor.excluded_*"],
+                "exclude_entities": ["binary_sensor.excluded"],
+            },
+            [
+                FilterTest("climate.excluded", False),
+                FilterTest("light.included", True),
+                FilterTest("sensor.excluded_test", False),
+                FilterTest("sensor.included_test", True),
+                FilterTest("binary_sensor.included", True),
+                FilterTest("binary_sensor.excluded", False),
+            ],
+        ),
+        (
+            {
+                "include_domains": ["light"],
+                "include_entity_globs": ["*.included_*"],
+                "exclude_domains": ["climate"],
+                "exclude_entity_globs": ["*.excluded_*"],
+                "exclude_entities": ["light.excluded"],
+            },
+            [
+                FilterTest("light.included", True),
+                FilterTest("light.excluded_test", False),
+                FilterTest("light.excluded", False),
+                FilterTest("sensor.included_test", True),
+                FilterTest("climate.included_test", False),
+            ],
+        ),
+        (
+            {
+                "include_entities": ["climate.included", "sensor.excluded_test"],
+                "exclude_domains": ["climate"],
+                "exclude_entity_globs": ["*.excluded_*"],
+                "exclude_entities": ["light.excluded"],
+            },
+            [
+                FilterTest("climate.excluded", False),
+                FilterTest("climate.included", True),
+                FilterTest("switch.excluded_test", False),
+                FilterTest("sensor.excluded_test", True),
+                FilterTest("light.excluded", False),
+                FilterTest("light.included", True),
+            ],
+        ),
+    ],
+    ids=["allowlist", "denylist", "filtered_allowlist", "filtered_denylist"],
+)
+async def test_filter(hass, entry, tests, mock_create_batch):
+    """Test different filters.
+
+    Filter_schema is also a fixture which is replaced by the filter_schema in the parametrize and added to the entry fixture.
+    """
+    for test in tests:
+        hass.states.async_set(test.entity_id, STATE_ON)
+        async_fire_time_changed(
+            hass, utcnow() + timedelta(seconds=entry.options[CONF_SEND_INTERVAL])
+        )
+        await hass.async_block_till_done()
+        if test.should_pass:
```

Comment:
Can we do the parametrization so we don't need any conditional logic in the test?

Reply:
yes, we can.